### PR TITLE
feat: add delete project frontend

### PR DESCRIPTION
**Pull request by rephus**
### Pre requirements:
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/lightdash/lightdash/blob/main/.github/CONTRIBUTING.md#sending-a-pull-request).

### Reference:
Closes:#1561

### Description:
Add frontend wrappers to delete projects

### Preview:

image

image

### Scope of the changes (select all that apply):
- [x] Frontend  
- [ ] Backend  
- [ ] Common  
- [ ] E2E tests  
- [ ] Docs  
- [ ] CI/CD

### CODE REVIEW

**Comment by rephus** [comment]
Ok now the frontend contaoins backend changes and merges into the backend

**Comment by rephus** [comment]
Mar-23-2022 13-15-15

**Comment by rephus** [comment]
Handle current project deletion 

Mar-23-2022 13-31-27

**Comment by rephus** [comment]
Updated modal message

image

File: packages/frontend/src/hooks/useProjects.ts [hunk 1]
```diff
@@ -25,3 +26,30 @@ export const useDefaultProject = () => {
...
+        url: `/org/projects/${id}`,
+        method: 'DELETE',
+        body: undefined,
+    });
+
+export const useDeleteProjectMutation = () => {
+    const queryClient = useQueryClient();
+    const { showToastSuccess, showToastError } = useApp();
+    return useMutation<undefined, ApiError, string>(deleteProjectQuery, {
+        mutationKey: ['saved_query_create'],
```
Comment (ZeRego) [inline packages/frontend/src/hooks/useProjects.ts:41]: We should change this key.

File: packages/frontend/src/components/UserSettingsModal/ProjectManagementPanel/index.tsx [hunk 2]
```diff
@@ -17,33 +26,87 @@ import {
...
+    );
+};
 
 const ProjectManagementPanel: FC = () => {
     const { data } = useProjects();
     const history = useHistory();
-    const params = useParams<{ projectUuid: string | undefined }>();
+    const params = useParams<{ projectUuid: string }>();
+
+    //const updateUser = useUpdateUserMutation(userUuid);
```
Comment (ZeRego) [inline packages/frontend/src/components/UserSettingsModal/ProjectManagementPanel/index.tsx:108]: forgotten comment :)

File: packages/frontend/src/components/UserSettingsModal/ProjectManagementPanel/index.tsx [hunk 3]
```diff
@@ -17,33 +26,87 @@ import {
...
+                </ProjectInfo>
+                <ButtonGroup>
+                    <LinkButton
+                        icon="cog"
+                        outlined
+                        text="Settings"
+                        href={`/projects/${projectUuid}/settings`}
+                    />
+                    <Button
+                        icon="trash"
```
Comment (ZeRego) [inline packages/frontend/src/components/UserSettingsModal/ProjectManagementPanel/index.tsx:51]: In the design and in the user management panel we use the `delete` icon. Maybe we should use it here to keep it consistent.

**Review by ZeRego — CHANGES_REQUESTED** [review]

**Review by ZeRego — APPROVED** [review]

**Pull request merged**